lightning-c-bindings/README.md

```diff
@@ -0,0 +1,193 @@
+   you to use the native Rust object in place of the trait. Such generated objects are only valid as
+   long as the original Rust native object is owned by a C-wrapped struct, and has not been `free`'d
+   or moved as a part of a Rust function call.
+
+ * Rust "unitary" enums are mapped simply as an equivalent C enum, however some Rust enums have
+   variants which contain payloads. Such enums are mapped automatically by cbindgen as a tag which
+   indicates the type and a union which holds the relevant fields for a given tag. A `X_free`
+   function is provided for the enum as a whole which automatically frees the correct fields for a
+   given tag, and a `Sentinel` tag is provided which causes the free function to do nothing (but
+   which must never appear in an enum when accessed by Rust code). The `Sentinel` tag is used by
+   the C++ wrapper classes to allow moving the ownership of an enum while invalidating the old copy.
+
+   For example, the unitary enum `LDKChannelMonitorUpdateErr` is mapped as follows:
+   ```
+   typedef enum LDKChannelMonitorUpdateErr {
+      /** .. */
+      LDKChannelMonitorUpdateErr_TemporaryFailure,
+      /** .. */
+      LDKChannelMonitorUpdateErr_PermanentFailure,
+      /** .. */
+      LDKChannelMonitorUpdateErr_Sentinel,
+   } LDKChannelMonitorUpdateErr;
+   ```
+
+   and the non-unitary enum LDKErrorAction is mapped as follows:
+   ```
+   typedef enum LDKErrorAction_Tag {
+      /** .. */
+      LDKErrorAction_DisconnectPeer,
+      /** .. */
+      LDKErrorAction_IgnoreError,
+      /** .. */
+      LDKErrorAction_SendErrorMessage,
+      /** .. */
+      LDKErrorAction_Sentinel,
+   } LDKErrorAction_Tag;
+
+   typedef struct LDKErrorAction_LDKDisconnectPeer_Body {
+      LDKErrorMessage msg;
+   } LDKErrorAction_LDKDisconnectPeer_Body;
+
+   typedef struct LDKErrorAction_LDKSendErrorMessage_Body {
+      LDKErrorMessage msg;
+   } LDKErrorAction_LDKSendErrorMessage_Body;
+
+   typedef struct LDKErrorAction {
+      LDKErrorAction_Tag tag;
+      union {
+         LDKErrorAction_LDKDisconnectPeer_Body disconnect_peer;
+         LDKErrorAction_LDKSendErrorMessage_Body send_error_message;
+      };
+   } LDKErrorAction;
+   ```
+
+ * Struct member functions are mapped as `Struct_function_name` and take a reference to the mapped
+   struct as their first argument. Free-standing functions are mapped simply as `function_name` and
+   take the relevant mapped type arguments.
+
+   Functions which return `&OpaqueRustType` and which return `OpaqueRustType` are both mapped to a
+   function returning an owned wrapper struct. The `_underlying_ref` flag (see above) will be set
+   to indicate that the pointed-to Rust object is owned or only a reference. Thus, when
+   implementing a function which Rust will call or calling a Rust function, you should check the
+   Rust documentation for the function to determine which behavior is expected.
```

Comment:
Does this mean: "Thus, when implementing a function which Rust will call or calling a Rust function, **and that function either takes in or returns a wrapper struct**, you should check the Rust documentation to determine whether an owned or a referenced object is parameterized/returned."

I mean, probably in any situation where you're implementing/calling a Rust function you should check the docs, but just want to make sure the `Thus` sentence actually logically follows haha, or if I'm misunderstanding something.